src/main.rs

```diff
@@ -347,27 +353,24 @@ impl FormatBamRecords {
         let mut spec = HashMap::new();
 
         for l in text.lines() {
-            match re.captures(l) {
-                Some(c) => {
-                    let mut read_spec = Vec::new();
-
-                    let read = c.get(1).unwrap().as_str().to_string();
-                    let tag_list = c.get(2).unwrap().as_str();
-                    let spec_elems = tag_list.split(',');
-                    for el in spec_elems {
-                        if el == "SEQ:QUAL" {
-                            read_spec.push(SpecEntry::Read)
-                        } else {
-                            let mut parts = el.split(':');
-                            let rtag = parts.next().unwrap().to_string();
-                            let qtag = parts.next().unwrap().to_string();
-                            read_spec.push(SpecEntry::Tags(rtag, qtag));
-                        }
+            if let Some(c) = re.captures(l) {
+                let mut read_spec = Vec::new();
+
+                let read = c.get(1).unwrap().as_str().to_string();
+                let tag_list = c.get(2).unwrap().as_str();
+                let spec_elems = tag_list.split(',');
+                for el in spec_elems {
+                    if el == "SEQ:QUAL" {
+                        read_spec.push(SpecEntry::Read)
+                    } else {
+                        let mut parts = el.split(':');
+                        let rtag = parts.next().unwrap().to_string();
+                        let qtag = parts.next().unwrap().to_string();
+                        read_spec.push(SpecEntry::Tags(rtag, qtag));
                     }
-
-                    spec.insert(read, read_spec);
                 }
-                None => (),
+
+                spec.insert(read, read_spec);
             }
```

Comment:
Every time I see `mut` I'm skeptical.  I think this loop would be better as
```suggestion
        let spec: HashMap<String, Vec<SpecEntry>> = text.lines().into_iter().filter_map(|l| {
            re.captures(l).map(|c| {
                let read = c.get(1).unwrap().as_str().to_string();
                let tag_list = c.get(2).unwrap().as_str();
                (read, tag_list.split(',').into_iter().map(|el| {
                    if el == "SEQ:QUAL" {
                       SpecEntry::Read
                    } else {
                        let mut parts = el.split(':');
                        let rtag = parts.next().unwrap().to_string();
                        let qtag = parts.next().unwrap().to_string();
                        SpecEntry::Tags(rtag, qtag)
                    }
                }
            })).collect())
        }).collect();
```

Or at the very least use `Vec::with_capacity()`.

Reply:
Indeed. I rewrote it a bit to avoid too much indentation when `rustfmt` reformatted it, but followed the overall idea.